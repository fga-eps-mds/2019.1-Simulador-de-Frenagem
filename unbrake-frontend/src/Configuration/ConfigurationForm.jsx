import React from "react";
import PropTypes from "prop-types";
import { initialize, Field, reduxForm } from "redux-form";
import { TextField, Checkbox } from "redux-form-material-ui";
import { withStyles, Button, FormControlLabel, Grid } from "@material-ui/core";
import styles from "./Styles";

const limits = (value, allValues) => {
  return parseInt(allValues.LSL, 10) >= parseInt(allValues.USL, 10)
    ? "Valores Inválidos"
    : undefined;
};

const validate = values => {
  const errors = {};
  const requiredFields = [
    "TAS",
    "TAT",
    "UWT",
    "NOS",
    "LSL",
    "USL",
    "TBS",
    "LWT"
  ];
  requiredFields.forEach(field => {
    if (!values[field]) {
      errors[field] = "Campo obrigatório";
    }
  });
  return errors;
};

const fieldsLabels = {
  NOS: "Numero de Snubs",
  USL: "Limite Superior (km/h)",
  UWT: "Tempo de Espera (s)",
  TBS: "Tempo entre ciclos",
  LSL: "Limite inferior (km/h)",
  LWT: "Tempo de espera (s)"
};

const rowsFields = (classes, vector, handleChange) => {
  let nameField;
  const fields = vector.map(value => {
    switch (value.name) {
      case "NOS":
        nameField = "NOS";
        break;
      case "USL":
        nameField = "USL";
        break;
      case "UWT":
        nameField = "UWT";
        break;
      case "TBS":
        nameField = "TBS";
        break;
      case "LSL":
        nameField = "LSL";
        break;
      case "LWT":
        nameField = "LWT";
        break;
      default:
        break;
    }
    return (
      <Grid key={`row${nameField}`} container xs={3} item justify="center">
        <div>
          <Field
            id={nameField}
            component={TextField}
            label={fieldsLabels[nameField]}
            onChange={handleChange}
            type="number"
            name={nameField}
            validate={nameField === "USL" || nameField === "LSL" ? limits : []}
            className={classes.textField}
            margin="normal"
            variant="outlined"
          />
        </div>
      </Grid>
    );
  });
  return fields;
};

const fieldsConfigurations = (classes, vector, handleChange) => {
  const rows = vector.map(value => {
    return (
      <Grid key={`${value[0].name}`} container xs={12} justify="center" item>
        {rowsFields(classes, value, handleChange)}
      </Grid>
    );
  });
  return rows;
};

const checkBox = (classes, type, handleChange) => {
  let label;
  switch (type.name) {
    case "TMO":
      label = "Inibe Desligamento do Motor";
      break;
    case "TAO":
      label = "Ativa saida auxiliar (AUX1)";
      break;
    default:
      break;
  }
  return (
    <Grid container item xs={3} style={{ paddingLeft: 20 }}>
      <FormControlLabel
        name={type.name}
        control={
          <Field
            component={Checkbox}
            onClick={handleChange}
            value={type.value}
          />
        }
        label={label}
      />
    </Grid>
  );
};

const CommunGrid = (classes, type, handleChange) => {
  let label;
  switch (type.name) {
    case "TAS":
      label = "Temperatura(˚C)(AUX1)";
      break;
    case "TAT":
      label = "Tempo (s)(AUX1)";
      break;
    default:
      break;
  }
  return (
    <Grid item container xs={3} className={classes.grid} justify="center">
      <Field
        id={type.name}
        component={TextField}
        label={label}
        value={type.value}
        onChange={handleChange}
        type="number"
        name={type.name}
        className={classes.textField}
        margin="normal"
        variant="outlined"
      />
    </Grid>
  );
};

const otherField = (classes, vector, handleChange) => {
  const fields = vector.map(value => {
    return (
      <Grid
        key={`other${value[1].name}`}
        container
        item
        xs={12}
        justify="center"
      >
        {checkBox(classes, value[0], handleChange)}
        {CommunGrid(classes, value[1], handleChange)}
        <Grid item container xs={3} />
      </Grid>
    );
  });
  return fields;
};

const Buttons = classes => {
  return (
    <Grid container item xs={3} className={classes.grid}>
      <Button type="submit" color="secondary" variant="contained">
        Cadastrar
      </Button>
    </Grid>
  );
};

const verifyCheckbox = variable => {
  let bool;
  if (variable === "FALSE" || variable === false) {
    bool = false;
  } else {
    bool = true;
  }
  return bool;
};

class ConfigurationForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      configuration: {
        NOS: "",
        USL: "",
        UWT: "",
        LSL: "",
        LWT: "",
        TBS: "",
        TAS: "",
        TAT: "",
        TMO: false,
        TAO: false
      }
    };
    this.handleChange = this.handleChange.bind(this);
  }

  shouldComponentUpdate(nextProps) {
    const { configuration, dispatch } = this.props;
    if (configuration !== nextProps.configuration) {
      const rightConfig = Object.assign({}, nextProps.configuration);
      const next = nextProps.configuration.CONFIG_ENSAIO;
      rightConfig.CONFIG_ENSAIO.TMO = verifyCheckbox(next.TMO);
      rightConfig.CONFIG_ENSAIO.TAO = verifyCheckbox(next.TAO);
      dispatch(initialize("configuration", rightConfig.CONFIG_ENSAIO));
      this.setState({ configuration: rightConfig.CONFIG_ENSAIO });
      return true;
    }
    return false;
  }

  handleChange(event) {
    const { target } = event;
    const value = target.type === "checkbox" ? target.checked : target.value;
    const configuration = { [event.target.name]: value };
    this.setState(prevState => ({
      configuration: { ...prevState.configuration, ...configuration }
    }));
  }

  render() {
    const { classes, handleSubmit, handleClickSave } = this.props;
    const { configuration } = this.state;
    const { TAS, TAT, TMO, TAO, UWT, NOS, LSL, USL, TBS, LWT } = configuration;
    const rowOne = [
      { name: "NOS", value: NOS },
      { name: "USL", value: USL },
      { name: "UWT", value: UWT }
    ];
    const rowTwo = [
      { name: "TBS", value: TBS },
      { name: "LSL", value: LSL },
      { name: "LWT", value: LWT }
    ];
    const rows = [rowOne, rowTwo];
    const oneFields = [
      { name: "TMO", value: TMO },
      { name: "TAS", value: TAS }
    ];  
    const twoFields = [
      { name: "TAO", value: TAO },
      { name: "TAT", value: TAT }
    ];
    const othersFields = [oneFields, twoFields];

    return (
      <form
        className={classes.container}
        onSubmit={handleSubmit(() => {
          handleClickSave(this.state);
        })}
      >
        {fieldsConfigurations(classes, rows, this.handleChange)}
        {otherField(classes, othersFields, this.handleChange)}
        <Grid container xs={12} item justify="center">
          <Grid
            item
            container
            xs={6}
            justify="center"
            className={classes.gridButton}
          >
            {Buttons(classes)}
          </Grid>
        </Grid>
      </form>
    );
  }
}

ConfigurationForm.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  configuration: PropTypes.oneOfType([PropTypes.object]).isRequired,
  dispatch: PropTypes.func.isRequired,
  handleClickSave: PropTypes.func.isRequired
};

const Configurationa = reduxForm({
  form: "configuration",
  validate
})(ConfigurationForm);

export default withStyles(styles)(Configurationa);
