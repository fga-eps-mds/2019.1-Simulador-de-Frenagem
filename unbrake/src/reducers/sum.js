const sum = (state = {}, action) => { //Jeito de definir uma função
  switch (action.type) {
    case 'ADD':
      console.log(state, action)
      return {...state, sum: +action.n1 + +action.n2};
      
    default:
      return state
  }
}

export default sum 
