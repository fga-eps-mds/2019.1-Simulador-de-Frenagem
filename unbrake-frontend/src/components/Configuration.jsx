import React from "react";
import iniparser from "iniparser";

import { Input, Grid, Button, Dialog } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";
import MenuItem from "@material-ui/core/MenuItem";
import TextField from "@material-ui/core/TextField";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Request from "../utils/Request";
import { API_URL_GRAPHQL } from "../utils/Constants";
import ConfigurationForm from "./ConfigurationForm";
import styles from "./Styles";

const query =
  "id, name, number, time, temperature, timeBetweenCycles, upperLimit, inferiorLimit, upperTime, inferiorTime, disableShutdown, enableOutput";

export async function submit(configuration, name) {
  if (name === "" || name === undefined) return;

  const url = `${API_URL_GRAPHQL}?query=mutation{createConfig(name:"${name}",number:${
    configuration.NOS
  },timeBetweenCycles:${configuration.TBS},upperLimit:${
    configuration.USL
  },inferiorLimit:${configuration.LSL},upperTime:${
    configuration.UWT
  },inferiorTime:${configuration.LWT},disableShutdown:${
    configuration.TMO
  },enableOutput:${configuration.TAO},temperature:${configuration.TAS},time:${
    configuration.TAT
  }){config{number, timeBetweenCycles,upperLimit,inferiorLimit}}}`;
  const method = "POST";
  await Request(url, method);

  window.location.reload();
}

const itensSelection = allConfiguration => {
  let allConfig = [{ id: 0, name: "" }];
  allConfig = allConfig.concat(allConfiguration);

  const itens = allConfig.map(value => {
    return (
      <MenuItem key={value.name + value.id} value={value.id}>
        {value.name}
      </MenuItem>
    );
  });
  return itens;
};

const selectConfiguration = (handleChange, configStates, classes) => {
  return (
    <Grid item xs={4} className={classes.title}>
      <TextField
        id="outlined-select-currency"
        select
        label="Configurações"
        value={configStates[0]}
        onChange={handleChange}
        name="dataBaseConfiguration"
        className={classes.formControl}
        margin="normal"
        variant="outlined"
      >
        {itensSelection(configStates[1])}
      </TextField>
    </Grid>
  );
};

const dialogName = (handleChange, handleClose, dialogStates) => {
  return (
    <Dialog
      open={dialogStates.open}
      onClose={handleClose}
      aria-labelledby="form-dialog-title"
    >
      <DialogTitle id="form-dialog-title">Nome da Configuração</DialogTitle>
      <DialogContent>
        <DialogContentText>
          Insira aqui o nome que você deseja dar para este arquivo de
          configuração
        </DialogContentText>
        <TextField
          autoFocus
          margin="dense"
          name="name"
          label="Nome"
          type="text"
          onChange={handleChange}
          value={dialogStates.name}
          fullWidth
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose} color="primary">
          Cancelar
        </Button>
        <Button
          onClick={() => submit(dialogStates.configuration, dialogStates.name)}
          color="primary"
        >
          Cadastrar
        </Button>
      </DialogActions>
    </Dialog>
  );
};

const createConfig = data => {
  const configurationDefault = {
    CONFIG_ENSAIO: {
      LSL: data.inferiorLimit,
      LWT: data.inferiorTime,
      NOS: data.number,
      TAO: data.enableOutput,
      TAS: data.temperature,
      TAT: data.time,
      TBS: data.timeBetweenCycles,
      TMO: data.disableShutdown,
      USL: data.upperLimit,
      UWT: data.upperTime
    }
  };
  return configurationDefault;
};

class Configuration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      configuration: {
        CONFIG_ENSAIO: {
          LSL: "",
          LWT: "",
          NOS: "",
          TAO: false,
          TAS: "",
          TAT: "",
          TBS: "",
          TMO: false,
          USL: "",
          UWT: ""
        },
        name: ""
      },
      fileName: "",
      dataBaseConfiguration: 0,
      allConfiguration: "",
      open: false
    };

    this.handleUpDefault = this.handleUpDefault.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.fileUpload = this.fileUpload.bind(this);
    this.handleClickOpen = this.handleClickOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleClickSave = this.handleClickSave.bind(this);
  }

  componentDidMount() {
    const url = `${API_URL_GRAPHQL}?query=query{configNotDefault{id, name}}`;
    const method = "GET";
    Request(url, method).then(json => {
      const data = json.data.configNotDefault;
      this.setState({ allConfiguration: data });
    });
  }

  handleClickSave(state) {
    const CONFIG_ENSAIO = state.configuration;
    const newConfig = { CONFIG_ENSAIO };
    this.setState({ configuration: newConfig });
    this.handleClickOpen();
  }

  handleClickOpen() {
    this.setState({ open: true });
  }

  handleClose() {
    this.setState({ open: false });
  }

  handleUpDefault() {
    const url = `${API_URL_GRAPHQL}?query=query{configDefault{${query}}}`;

    const method = "GET";

    const VAZIO = 0;

    Request(url, method).then(response => {
      if (response.data.configDefault.length === VAZIO) {
        return;
      }
      const data = response.data.configDefault[0];
      const configurationDefault = createConfig(data);
      this.setState({ configuration: configurationDefault });
    });
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
    const invalidId = 0;
    if (
      event.target.name === "dataBaseConfiguration" &&
      event.target.value > invalidId
    )
      this.handleSelectConfig(event.target.value);
  }

  handleSelectConfig(id) {
    const url = `${API_URL_GRAPHQL}?query=query{configAt(id:${id}){${query}}}`;

    const method = "GET";

    Request(url, method).then(response => {
      const data = response.data.configAt;

      const configurationDefault = createConfig(data);
      this.setState({ configuration: configurationDefault });
    });
  }

  uploadField(field) {
    const { classes } = this.props;
    const { fileName } = this.state;

    let archive;

    if (field === "calibration") archive = "Calibração";
    else archive = "Configuração";

    return (
      <Grid container item xs={10} alignItems="center" justify="center">
        <Grid item xs={4} className={classes.title}>
          <h2>Upload arquivo de {archive}</h2>
        </Grid>

        <Grid item xs={4} className={classes.grid}>
          <label htmlFor="contained-button-file">
            <input
              id="contained-button-file"
              type="file"
              name={field}
              className={classes.input}
              onChange={e => this.fileUpload(e.target.files[0], field)}
            />
            <Paper className={classes.rootUploadFile}>
              <IconButton component="span">
                <CloudUploadIcon style={{ color: "black" }} />
              </IconButton>
              <InputBase
                disabled
                className={classes.input_file_name}
                value={fileName}
                placeholder="Upload do arquivo de configuração"
              />
            </Paper>
          </label>
        </Grid>
      </Grid>
    );
  }

  fileUpload(file, name) {
    this.setState({ fileName: file.name });
    const formData = new FormData();
    formData.append("file", name);
    const reader = new FileReader();
    const scope = this;

    reader.onload = e => {
      const content = e.target.result;
      const fileUpload = iniparser.parseString(content);
      scope.setState({ configuration: fileUpload });
    };

    reader.readAsText(file, "UTF-8");
  }

  render() {
    const { classes } = this.props;
    const {
      configuration,
      dataBaseConfiguration,
      allConfiguration,
      open,
      name
    } = this.state;
    const configStates = [dataBaseConfiguration, allConfiguration];
    const dialogStates = {
      configuration: configuration.CONFIG_ENSAIO,
      open,
      name
    };

    return (
      <Grid alignItems="center" container className={classes.configuration}>
        <div>
          <Grid container item justify="center">
            {this.uploadField("configuration")}
          </Grid>
          <Grid container justify="center" item alignItems="center" xs={12}>
            {selectConfiguration(this.handleChange, configStates, classes)}
            <Button
              onClick={this.handleUpDefault}
              color="secondary"
              variant="contained"
            >
              Configuração Padrão
            </Button>
          </Grid>
          <Grid
            container
            alignItems="center"
            xs={12}
            item
            className={classes.form}
          >
            <ConfigurationForm
              configuration={configuration}
              handleClickSave={this.handleClickSave}
            />
          </Grid>
        </div>
        {dialogName(this.handleChange, this.handleClose, dialogStates)}
      </Grid>
    );
  }
}

Configuration.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired
};

export default withStyles(styles)(Configuration);
