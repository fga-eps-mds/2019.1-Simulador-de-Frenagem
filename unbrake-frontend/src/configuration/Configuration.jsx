import React from "react";
import iniparser from "iniparser";
import Paper from "@material-ui/core/Paper";
import { Grid, Button, Dialog } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";
import MenuItem from "@material-ui/core/MenuItem";
import TextField from "@material-ui/core/TextField";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import IconButton from "@material-ui/core/IconButton";
import CloudUploadIcon from "@material-ui/icons/CloudUpload";
import { connect } from "react-redux";
import Request from "../utils/Request";
import { API_URL_GRAPHQL } from "../utils/Constants";
import ConfigurationForm from "./ConfigurationForm";
import styles from "./Styles";
import { createConfig, query, submit } from "./ConfigFunctions";
import NotificationContainer from "../components/Notification";
import { messageSistem } from "../actions/NotificationActions";

const itensSelection = allConfiguration => {
  let allConfig = [{ id: 0, name: "" }];
  allConfig = allConfig.concat(allConfiguration);

  const itens = allConfig.map(value => {
    return (
      <MenuItem key={value.name + value.id} value={value.id}>
        {value.name}
      </MenuItem>
    );
  });
  return itens;
};

const selectConfiguration = (handleChange, configStates, classes) => {
  return (
    <Grid item xs={4} className={classes.title}>
      <TextField
        id="outlined-select-currency"
        select
        label="Configurações"
        value={configStates[0]}
        onChange={handleChange}
        name="dataBaseConfiguration"
        className={classes.formControl}
        margin="normal"
        variant="outlined"
      >
        {itensSelection(configStates[1])}
      </TextField>
    </Grid>
  );
};

const dialogName = (functions, sendMessage, dialogStates) => {
  return (
    <Dialog
      open={dialogStates.open}
      onClose={functions.handleClose}
      aria-labelledby="form-dialog-title"
    >
      <DialogTitle id="form-dialog-title">Nome da Configuração</DialogTitle>
      <DialogContent>
        <DialogContentText>
          Insira aqui o nome que você deseja dar para este arquivo de
          configuração
        </DialogContentText>
        <TextField
          autoFocus
          margin="dense"
          name="name"
          label="Nome"
          type="text"
          onChange={functions.handleChange}
          value={dialogStates.name}
          fullWidth
        />
      </DialogContent>
      <DialogActions>
        <Button onClick={functions.handleClose} color="primary">
          Cancelar
        </Button>
        <Button
          onClick={() =>
            submit(dialogStates.configuration, dialogStates.name, sendMessage)
          }
          color="primary"
        >
          Cadastrar
        </Button>
      </DialogActions>
    </Dialog>
  );
};

const defaultButton = handleUpDefault => {
  return (
    <Button onClick={handleUpDefault} color="secondary" variant="contained">
      Configuração Padrão
    </Button>
  );
};

class Configuration extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      configuration: {
        CONFIG_ENSAIO: {
          LSL: "",
          LWT: "",
          NOS: "",
          TAO: false,
          TAS: "",
          TAT: "",
          TBS: "",
          TMO: false,
          USL: "",
          UWT: ""
        },
        name: ""
      },
      fileName: "Upload do arquivo de configuração",
      dataBaseConfiguration: 0,
      allConfiguration: "",
      open: false
    };

    this.handleUpDefault = this.handleUpDefault.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.fileUpload = this.fileUpload.bind(this);
    this.handleClickOpen = this.handleClickOpen.bind(this);
    this.handleClose = this.handleClose.bind(this);
    this.handleClickSave = this.handleClickSave.bind(this);
  }

  componentDidMount() {
    const url = `${API_URL_GRAPHQL}?query=query{configNotDefault{id, name}}`;
    const method = "GET";
    Request(url, method).then(json => {
      const data = json.data.configNotDefault;
      if (data !== null) this.setState({ allConfiguration: data });
    });
  }

  handleClickSave(state) {
    const CONFIG_ENSAIO = state.configuration;
    const newConfig = { CONFIG_ENSAIO };
    this.setState({ configuration: newConfig });
    this.handleClickOpen();
  }

  handleClickOpen() {
    this.setState({ open: true });
  }

  handleClose() {
    this.setState({ open: false });
  }

  handleUpDefault() {
    const url = `${API_URL_GRAPHQL}?query=query{configDefault{${query}}}`;

    const method = "GET";

    const VAZIO = 0;

    Request(url, method).then(response => {
      if (response.data.configDefault.length === VAZIO) {
        return;
      }
      const data = response.data.configDefault[0];
      const configurationDefault = createConfig(data);
      this.setState({ configuration: configurationDefault });
    });
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
    const invalidId = 0;
    if (
      event.target.name === "dataBaseConfiguration" &&
      event.target.value > invalidId
    )
      this.handleSelectConfig(event.target.value);
  }

  handleSelectConfig(id) {
    const url = `${API_URL_GRAPHQL}?query=query{configAt(id:${id}){${query}}}`;

    const method = "GET";

    Request(url, method).then(response => {
      const data = response.data.configAt;

      const configurationDefault = createConfig(data);
      this.setState({ configuration: configurationDefault });
    });
  }

  uploadField(field) {
    const { classes } = this.props;
    const { fileName } = this.state;

    let archive;

    if (field === "calibration") archive = "Calibração";
    else archive = "Configuração";

    return (
      <Grid container item xs={10} alignItems="center" justify="center">
        <Grid item xs={4} className={classes.title}>
          <h2>Upload arquivo de {archive}</h2>
        </Grid>

        <Grid item xs={4} className={classes.grid}>
          <label htmlFor="contained-button-file">
            <input
              id="contained-button-file"
              type="file"
              name={field}
              className={classes.input}
              onChange={e => this.fileUpload(e.target.files[0], field)}
            />
            <Paper className={classes.rootUploadFile}>
              <IconButton component="span">
                <CloudUploadIcon style={{ color: "black" }} />
              </IconButton>
              <span
                className={classes.input_file_name}
                placeholder="Upload do arquivo de configuração"
              >
                {fileName}
              </span>
            </Paper>
          </label>
        </Grid>
      </Grid>
    );
  }

  fileUpload(file, name) {
    const { sendMessage } = this.props;
    this.setState({ fileName: file.name });
    const formData = new FormData();
    formData.append("file", name);
    const reader = new FileReader();
    const scope = this;

    reader.onload = e => {
      const content = e.target.result;
      const fileUpload = iniparser.parseString(content);
      if (fileUpload.CONFIG_ENSAIO !== undefined)
        scope.setState({ configuration: fileUpload });
      else {
        sendMessage({
          message: "Aquivo não se encontra no padrão .ini para configuração",
          variante: "warning",
          condition: true
        });
      }
    };

    reader.readAsText(file, "UTF-8");
  }

  render() {
    const { classes, sendMessage } = this.props;
    const {
      configuration,
      dataBaseConfiguration,
      allConfiguration,
      open,
      name
    } = this.state;
    const functions = {
      handleChange: this.handleChange,
      handleClose: this.handleClose
    };
    const configStates = [dataBaseConfiguration, allConfiguration];
    const dialogStates = {
      configuration: configuration.CONFIG_ENSAIO,
      open,
      name
    };

    return (
      <Grid alignItems="center" container className={classes.configuration}>
        <div>
          <Grid container item justify="center">
            {this.uploadField("configuration")}
          </Grid>
          <Grid container justify="center" item alignItems="center" xs={12}>
            {selectConfiguration(this.handleChange, configStates, classes)}
            {defaultButton(this.handleUpDefault)}
          </Grid>
          <Grid
            container
            alignItems="center"
            xs={12}
            item
            className={classes.form}
          >
            <ConfigurationForm
              configuration={configuration}
              handleClickSave={this.handleClickSave}
            />
          </Grid>
        </div>
        {dialogName(functions, sendMessage, dialogStates)}
        <NotificationContainer />
      </Grid>
    );
  }
}

Configuration.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  sendMessage: PropTypes.func.isRequired
};

const mapDispatchToProps = dispatch => ({
  sendMessage: payload => dispatch(messageSistem(payload))
});

export default connect(
  null,
  mapDispatchToProps
)(withStyles(styles)(Configuration));
