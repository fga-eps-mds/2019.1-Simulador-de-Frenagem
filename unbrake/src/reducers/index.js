import { combineReducers } from 'redux'
import sum from './sum'

const Reducer = combineReducers({ //Agrupa os componentes
	sum,
})

export default Reducer;