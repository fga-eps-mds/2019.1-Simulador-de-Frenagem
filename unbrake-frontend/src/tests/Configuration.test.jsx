import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createStore } from "redux";
import { Provider } from "react-redux";
import Configuration from "../Configuration/Configuration";
import ComponentTest from "./ComponentTest";
import reducers from "../reducer/index";

/*
 * import fetch from "jest-fetch-mock";
 * import { API_URL_GRAPHQL } from "../utils/Constants";
 */

Enzyme.configure({ adapter: new Adapter() });

describe("<Configuration />", () => {
  const context = React.createContext();
  const store = createStore(reducers);
  const wrapper = shallow(
    <Provider context={context} store={store}>
      <Configuration context={context} />
    </Provider>
  );

  ComponentTest(wrapper);

  /*
   * test("submit()", () => {
   *   const url
   */

  // })
});

/*
 * describe("submit", () => {
 *   it("Call request and return response", () => {
 *     beforeEach(() => {
 *       fetch.resetMocks();
 *     });
 *     const configuration = {
 *       LSL: 10,
 *       LWT: 10,
 *       NOS: 10,
 *       TAO: false,
 *       TAS: 10,
 *       TAT: 10,
 *       TBS: 10,
 *       TMO: false,
 *       USL: 10,
 *       UWT: 10
 *     };
 *     const name = "test";
 *     let response = "";
 *     fetch.mockResponseOnce(
 *       JSON.stringify({ data: { createConfig: { name } } })
 *     );
 */

/*
 *     submit(configuration, name)
 *       .then(res => {
 *         response = res.data.createConfig.name;
 *         expect(response).toBe(name);
 *         expect(fetch.mock.calls[0][0]).toEqual(API_URL_GRAPHQL);
 *       })
 *       .catch(() => {});
 *   });
 * });
 */
