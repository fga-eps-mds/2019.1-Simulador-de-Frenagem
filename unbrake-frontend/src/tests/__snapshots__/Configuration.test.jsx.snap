// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Configuration /> render() renders the component 1`] = `
<WithStyles(Grid)
  alignItems="center"
  container={true}
  justify="center"
  spacing={40}
  style={
    Object {
      "minHeight": "5px",
    }
  }
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    item={true}
    justify="center"
    xs={10}
  >
    <WithStyles(Grid)
      className="Configuration-title-1"
      item={true}
      xs={4}
    >
      <h2
        justify="left"
      >
        Upload arquivo de 
        Configuração
      </h2>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      className="Configuration-grid-2"
      item={true}
      xs={4}
    >
      <WithStyles(Input)
        name="configuration"
        onChange={[Function]}
        type="file"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    className="Configuration-title-1"
    item={true}
    xs={4}
  >
    <WithStyles(FormControl)
      className="Configuration-formControl-3"
      variant="outlined"
    >
      <WithStyles(WithFormControlContext(InputLabel))
        htmlFor="outlined-age-simple"
      >
        Configurações
      </WithStyles(WithFormControlContext(InputLabel))>
      <WithStyles(WithFormControlContext(Select))
        input={
          <WithStyles(OutlinedInput)
            id="outlined-age-simple"
            labelWidth={0}
            name="dataBaseConfiguration"
          />
        }
        onChange={[Function]}
        value={0}
      >
        <WithStyles(MenuItem)
          key="0"
          value={0}
        />
      </WithStyles(WithFormControlContext(Select))>
    </WithStyles(FormControl)>
  </WithStyles(Grid)>
  <WithStyles(ReduxForm)
    configuration={
      Object {
        "CONFIG_ENSAIO": Object {
          "LSL": "",
          "LWT": "",
          "NOS": "",
          "TAO": false,
          "TAS": "",
          "TAT": "",
          "TBS": "",
          "TMO": false,
          "USL": "",
          "UWT": "",
        },
      }
    }
  />
  <WithStyles(Button)
    color="secondary"
    onClick={[Function]}
    variant="contained"
  >
    Padrão
  </WithStyles(Button)>
</WithStyles(Grid)>
`;
